server/lib/src/idm/ldap.rs

```diff
@@ -536,9 +625,39 @@ impl LdapServer {
                 // No need to notify on unbind (per rfc4511)
                 Ok(LdapResponseState::Unbind)
             }
-            ServerOps::Compare(cr) => Ok(LdapResponseState::Respond(
-                cr.gen_error(LdapResultCode::Other, "not supported".to_string()),
-            )),
+            ServerOps::Compare(cr) => match uat {
+                Some(u) => self
+                    .do_compare(idms, &cr, &u, source)
+                    .await
+                    .map(LdapResponseState::MultiPartResponse)
+                    .or_else(|e| {
+                        let (rc, msg) = operationerr_to_ldapresultcode(e);
+                        Ok(LdapResponseState::Respond(cr.gen_error(rc, msg)))
+                    }),
+                None => {
+                    // Compare can occur without a bind, so bind first.
```

Comment:
Might be worth adding a comment here that this is per section 4 of rfc 4513 ( https://www.rfc-editor.org/rfc/rfc4513#section-4 )

Reply:
added a comment to both operations.